src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyProfileServiceImplTest.java

```diff
@@ -47,56 +59,143 @@ class CompanyProfileServiceImplTest {
+                .thenReturn(Optional.empty());
+
+        assertFalse(this.companyProfileService.delete(COMPANY_NUMBER));
+        verify(repository, never()).delete(any());
+    }
+
+    // Test that a company profile is created with dissolved company status
+    @Test
+    void createDissolvedCompany() {
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyStatus(COMPANY_STATUS_DISSOLVED);
+        assertCompanyProfile( COMPANY_STATUS_DISSOLVED, Jurisdiction.ENGLAND_WALES.toString(), COMPANY_TYPE_LTD, true);
+    }
+
+    // Test that a company profile is created with plc company type
+    @Test
+    void createPlcCompany() {
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyType(COMPANY_TYPE_PLC);
+        assertCompanyProfile( COMPANY_STATUS_ACTIVE, Jurisdiction.ENGLAND_WALES.toString(), COMPANY_TYPE_PLC, false);
+    }
+
+    @Test
+    void testInvalidCompanyStatus() {
+        validator = factory.getValidator();
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyStatus("invalid-company-status");
+        Set<ConstraintViolation<CompanySpec>> violations = validator.validate(spec);
+        assertTrue(violations.stream().anyMatch(v -> "Invalid company status".equals(v.getMessage())), "Expected a violation message for invalid company status");
+    }
+
+    @Test
+    void testInvalidCompanyType() {
+        validator = factory.getValidator();
+        spec.setJurisdiction(Jurisdiction.ENGLAND_WALES);
+        spec.setCompanyType("invalid-company-type");
+        Set<ConstraintViolation<CompanySpec>> violations = validator.validate(spec);
+        assertTrue(violations.stream().anyMatch(v -> "Invalid company type".equals(v.getMessage())), "Expected a violation message for invalid company type");
+    }
+
+    // Test that a spec is not accepting field attributes that are not defined in the CompanySpec class
+    @Test
+    void testDeserializationFailsOnUnknownProperties() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String invalidJson = "{ \"jurisdiction\": \"england-wales\", \"companystatus\": \"active\" }";
+        assertThrows(UnrecognizedPropertyException.class, () -> {
+            objectMapper.readValue(invalidJson, CompanySpec.class);
+        });
+    }
+
+    // Test that a spec is accepting field attributes that are defined in the CompanySpec class
+    @Test
+    void testDeserializationSucceedsWithValidProperties() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String validJson = "{ \"jurisdiction\": \"england-wales\", \"company_status\": \"dissolved\" }";
+        assertDoesNotThrow(() -> {
+            objectMapper.readValue(validJson, CompanySpec.class);
+        });
+    }
+
+    private void assertCompanyProfile(String companyStatus, String jurisdiction, String companyType, Boolean hasInsolvencyHistory) {
```

Comment:
I would rename to reflect this is calling `create`

Reply:
renamed the method name

Reply:
Maybe we should keep the "assert" bit? How about `assertCreateCompanyProfile`?